packages/datatrak-web/src/api/mutations/useLogout.ts

```diff
@@ -1,12 +1,17 @@
 import { useMutation, useQueryClient } from '@tanstack/react-query';
 import { post } from '../api';
+import { useDatabaseContext } from '../../hooks/database';
+import { clearDatabase } from '../../database';
 
 export const useLogout = () => {
   const queryClient = useQueryClient();
+  const { models } = useDatabaseContext();
 
   return useMutation(['logout'], () => post('logout'), {
     onSuccess: async () => {
       await queryClient.resetQueries();
+
+      await clearDatabase(models);
```

Comment:
Should this happen immediately upon login? What happens if the same user logs in again?

- If online, it just re-syncs, which is maybe fine. Costs time, but maybe gives users a way to reset their DB state.
- If offline, are they forced to go back online to login? This could make accidental logouts quite frustrating

Does it make sense to clear the DB only when a **different** user has successfully logged in?

***

If keeping as-is, then:

```suggestion
      await Promise.all([queryClient.resetQueries(), clearDatabase(models)]);
```